### ISSUE TITLE
Issue in gun_actor.  Feral security guards cannot miss

### ISSUE THREAD

**Issue by Maleclypse** [CleverRaven/Cataclysm-DDA#51892]
### Describe the bug

Feral security guards cannot miss.  They can only miss if they hit something between you and them.   Ties into 
#51825.  Feral Guard gun skills 
```
    "special_attacks": [
      {
        "type": "gun",
        "cooldown": 10,
        "move_cost": 150,
        "gun_type": "m9",
        "ammo_type": "9mm",
        "fake_skills": [ [ "gun", 1 ], [ "handgun", 2 ] ],
        "fake_dex": 8,
        "fake_per": 10,
        "ranges": [ [ 0, 14, "DEFAULT" ] ],
        "require_targeting_player": false,
        "description": "The feral security guard fires their Beretta M9A1!"
      }
    ],
```

### Steps To Reproduce

Make new character. 
spawn horde of 14 security guards around avatar but not in melee range
pause for one turn.
Every security guard in 14 squares of PC will fire and every one of those shots will hit either the PC or a guard in between the shooter and the PC.

Test with modified special attack
```
    "special_attacks": [
      {
        "type": "gun",
        "cooldown": 10,
        "move_cost": 150,
        "gun_type": "m9",
        "ammo_type": "9mm",
        "fake_skills": [ [ "handgun", 1 ] ],
        "fake_dex": 4,
        "fake_per": 6,
        "ranges": [ [ 0, 14, "DEFAULT" ] ],
        "require_targeting_player": false,
        "description": "The feral security guard fires their Beretta M9A1!"
      }
    ],
```
Same results, no misses.

### Expected behavior

Ferals should mostly miss each shot.  I'll try shrinking their skills in JSON and report back here 

### Screenshots

![9mm](https://user-images.githubusercontent.com/54345792/134790747-cfdcf041-29d2-4cd2-9e7b-93803ac4ebc7.PNG)
Current stats and gunfire.

Test with modified special attack
![lowered feral skills](https://user-images.githubusercontent.com/54345792/134790919-4eeee109-d2f1-4deb-a303-aa9ae35cda16.png)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0 2009
- Game Version: 20bf2de [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Bionic Professions [package_bionic_professions]
]

**Comment by RoyBerube** [comment]
"handgun" skill in the feral guard json should actually be "pistol", which is the skill required by their weapon. Change it to pistol and they are less accurate. 

Without pistol defined in json, the pistol skill defaults to 8. That is why they are so accurate.

**Comment by Maleclypse** [comment]
Closed by #51916
